src/patterns/ec2-app/base.ts

```diff
@@ -158,176 +194,11 @@ function restrictedCidrRanges(ranges: IPeer[]) {
 }
 
 /**
- * Pattern which creates the resources needed to run an application on EC2 instances.
- * For convenience, you may wish to use [[`GuPlayApp`]] or [[`GuNodeApp`]], which extend this class.
- *
- * This pattern will grant your EC2 instances a number of commonly needed IAM permissions. For more information on
- * this, see [[`GuInstanceRole`]]. To add additional permissions to your EC2 instances, see [[`GuEc2AppProps`]].
- *
```

Comment:
This change removes some debatably useful documentation here, (though I agree this might not be the best place for these examples). Where is the best place for examples of usage of these patterns?

Reply:
I'm not sure - I wonder if we should add some methods on patterns too, a la CDK itself, to support this kind of thing. E.g. something like this (for a function example):

```
public addToRolePolicy(statement: PolicyStatement): void {
		this.function.addToRolePolicy(statement);
	}
```

I experimented a bit with this here anyway, as it felt nice:

https://github.com/guardian/cdk-v2/blob/main/lib/lambda-task.ts#L159